### PR TITLE
fix(ui): fix NaN input in Pagination

### LINKED ISSUE #1793: [Bug](ui): when NaN input is entered into the Pagination field, NaN in printed
**Description**

When NaN input is entered into the Pagination field e.g "#", NaN in printed

Image

**Steps to Reproduce**

1. Enter a NaN value e.g. "#"
2. Notice NaN is printed in the input field

**Expected Behavior**

Default to the last available page.

**Environment**

- OS: [e.g., Windows, macOS, Linux]
- Browser: [e.g., Chrome, Firefox, Safari]
- Version: [e.g., 22]

### PR DESCRIPTION
**Pull request by guoda-puidokaite**
# Summary

When NaN input is entered, instead of printing `NaN`, the last available page is defaulted to.

# Related Issues

Closes https://github.com/cloudoperators/juno/issues/1793

# Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have made corresponding changes to the documentation (if applicable).
- [ ] My changes generate no new warnings or errors.
- [ ] I have created a changeset for my changes.

# PR Manifesto

Review the [PR Manifesto](https://github.com/cloudoperators/juno/blob/main/docs/pr_manifesto.md) for best practises.

### CODE REVIEW

File: packages/ui-components/src/components/Pagination/Pagination.component.tsx [hunk 1]
```diff
@@ -184,7 +184,7 @@ export const Pagination = ({
       // Enforce minimum and maximum limits
       if (inputValue < 1) {
         inputValue = 1
-      } else if (controlTotalPage !== undefined && inputValue > controlTotalPage) {
+      } else if (isNaN(inputValue) || (controlTotalPage !== undefined && inputValue > controlTotalPage)) {
```
Comment (edda) [inline packages/ui-components/src/components/Pagination/Pagination.component.tsx:187]: hmm, this won't work if `controlTotalPage` is `undefined`. Imo a better way to deal with `NaN` values is to add it to the `inputValue < 1` case instead, because there is always going to be a first page.

**Review by ArtieReus — COMMENTED** [review]
what about  adding following tests: 
  - What happens when totalPages is undefined and user enters non-numeric input
  - Edge cases like empty string, whitespace, special characters beyond just "$"

The test only covers the case where totalPages={5} is defined.

File: packages/ui-components/src/components/Pagination/Pagination.component.tsx (hunk 1 above)
Comment (TilmanHaupt) [reply packages/ui-components/src/components/Pagination/Pagination.component.tsx:187]: !Number.isInteger(inputValue)  - is maybe also the wanted function to catch everything what is not a possible page. As 0.5 is not a possible page.
